### PR TITLE
I want to view details of my purchases to better understand my purchase patterns #14

### PR DESCRIPTION
**Pull request by AshlingH**
## Description

This code creates an exciting feature. The user is able to click on a purchased item, a modal will show the item details on screen.

## Related Issue
#13 

## Acceptance Criteria

- From the shopping list, user can tap to navigate to a detailed view of the item

- Detail view includes the following information:
     - Name of the item
     - Last purchased date
     - Estimated number of days until the next purchase
     - Number of times the item has been purchased"

- When in the “item details” view, a back arrow should appear in the header that takes the user back to the “list” view

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
| ✓    | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates
Creating New modal to show item details so user is better able to make decisions of when to next purcahse the item.

### CODE REVIEW

**Review by jonbascos — COMMENTED** [review]
Looks awesome!  I really like the unpacking of the props too.  That's something I should start doing.  It makes more sense than just putting in the word 'props'.  Looks clean!

File: src/lib/timeframeConstants.js [hunk 1]
```diff
@@ -1,6 +1,30 @@
...
+        output=returnStringDict[number];
+    }
+    if(type === "class"){
+        output=returnClassStringDict[number];
+    }
+    return output;
+}
+
+
+export {SOON, KIND_OF_SOON, NOT_SOON, INACTIVE, returnString};
```
Comment (laurenmbeatty) [inline src/lib/timeframeConstants.js:30]: Love this file! Well done....so handy!

File: src/pages/ShoppingList.js [hunk 2]
```diff
@@ -191,16 +204,17 @@ const ShoppingList = ({ token }) => {
                 onChange={e => setFilterString(e.target.value)}
             />
             <button onClick={() => setFilterString('')}>X</button>
-            <table>
+            <tbody>
                 {filterString
```
Comment (laurenmbeatty) [inline src/pages/ShoppingList.js:208]: This `tbody` should be wrapped in a `<table>` tag, too. Then you will get the table styles back:
image

image

File: src/components/ShoppingListItem.js [hunk 3]
```diff
@@ -57,13 +17,13 @@ const ShoppingListItem = ({item, handleCheck, setCurrentItem, setModal}) => {
                 onChange={e => handleCheck(e, item)}
                 />
             </td>
-            <td>{item.itemName}</td>
-            <td>{timeFrameString(item.timeFrame)}</td>
+            <td onClick= {()=> {setCurrentItem(item); setDetailModal(true);}} style={{cursor: "pointer"}}>{item.itemName}</td>
+            <td>{returnString("value",item.timeFrame)}</td>
```
Comment (dkimlim) [inline src/components/ShoppingListItem.js:21]: yeyy! So nice you were able to refactor this

File: src/lib/timeframeConstants.js [hunk 4]
```diff
@@ -1,6 +1,30 @@
...
+returnClassStringDict[KIND_OF_SOON.integerValue] = KIND_OF_SOON.stringClass;
+returnClassStringDict[NOT_SOON.integerValue] =  NOT_SOON.stringClass;
+returnClassStringDict[INACTIVE.integerValue] = INACTIVE.stringClass;
+
+const returnString = (type, number) => {
+    let output;
+    if(type === "value"){
+        output=returnStringDict[number];
+    }
+    if(type === "class"){
```
Comment (dkimlim) [inline src/lib/timeframeConstants.js:23]: cool little function :)

**Review by dkimlim — APPROVED** [review]
SOOOO good reading your PR this week! Good job ya'll, it's looking squeaky clean

File: src/components/Modal.js [hunk 5]
```diff
@@ -1,14 +1,53 @@
...
+                <p> Are you sure you want to delete {item.itemName}?</p>
+                <button onClick={() => deleteItem(item)}>YES</button>
+                <button onClick={() => cancelItem()}>No</button>
+            </div>
+        );
+    } else if (type === 'detail') {
+        output = (
+            <div className="detailsModal">
+                <h1>Purchase Details</h1>
+                <h2>{item.name}</h2>
```
Comment (laurenmbeatty) [inline src/components/Modal.js]:
```suggestion
                <h2>{item.itemName}</h2>
```
The name isn't showing up in the modal...I think it needs to be `item.itemName`

**Review by laurenmbeatty — CHANGES_REQUESTED** [review]
This looks fantastic. Just one tiny bug with the item name in the modal. One thing that would be a "nice-to-have"...could you make it so if I have the details modal open, and then click the delete button, that the details modal closes? Same thing if I have the delete modal open, and then click the details modal...could the delete modal close?

**Review by laurenmbeatty — APPROVED** [review]
Nice!  Thanks for the change @AshlingH !  This looks great.

**Pull request merged**